react/ProductSummaryList.js

```diff
@@ -0,0 +1,193 @@
+import React, { createContext, useContext } from 'react'
+import { compose, graphql } from 'react-apollo'
+import { injectIntl } from 'react-intl'
+import { formatIOMessage } from 'vtex.native-types'
+
+import ProductSummary from './components/ProductSummary'
+import ProductSummaryName from './components/ProductSummaryName/ProductSummaryName'
+import ProductSummaryPrice from './components/ProductSummaryPrice/ProductSummaryPrice'
+import ProductSummaryBuyButton from './components/ProductSummaryBuyButton/ProductSummaryBuyButton'
+import ProductSummaryImage from './components/ProductSummaryImage/ProductImage'
+import Spacer from './Spacer'
+import { mapCatalogProductToProductSummary } from './utils/normalize'
+
+import productsQuery from './graphql/products.graphql'
+
+const ProductSummaryListStateContext = createContext(undefined)
+
+const ProductSummaryList = ({ children, data, intl, buyButtonText }) => {
+  const formattedBuyButtonText = formatIOMessage({ id: buyButtonText, intl })
+  const componentList = data.products.map(product => {
+    const normalizedProduct = mapCatalogProductToProductSummary(product)
+
+    return (
+      <ProductSummary key={product.productId} product={normalizedProduct}>
+        <ProductSummaryImage />
+        <ProductSummaryName />
```

Comment:
Can't this point to an extension point so this layout may be flexible?

Reply:
It did work the way I tried to! I'm not sure how to make this work with `<ExtensionPoint />`, because if I just pass the id `"product-summary"`, or `product-summary.shelf`, for example, it renders nothing, even when these blocks are defined in the store's theme :(
It would surely be ideal!

Reply:
We can't go back to the old ways of fixed layout. We need to figure this out.